### PR TITLE
Node credential button fix

### PR DESCRIPTION
**Pull request by shivangi-singh1**
Signed-off-by: Shivangi Singh <shisingh@progress.com>

### :nut_and_bolt: Description: What code changed, and why?
Bug : 1. Save button for 'Reset credentials' for the node credential tab should work fine for all -: sudo, ssh and winrm.
2.Save button css should be changed.

### :chains: Related Resources
fix button issue and loading when we change cred for second time in node credential tab : #6078 

### :+1: Definition of Done
1. Save button for 'Reset credentials' for the node credential tab should work fine for all -: sudo, ssh and winrm.
2.Save button css for reset credentials should be corrected.

### :athletic_shoe: How to Build and Test the Change

```
STEP 1
inside the hab studio

[default:/src:0]# build components/automate-ui-devproxy/
[default:/src:0]# start_automate_ui_background
[default:/src:0]# start_all_services

STEP 2
open new window
go to automate UI path

$ cd components/automate-ui
and run the command 

npm run serve:hab

```

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable

https://user-images.githubusercontent.com/87964433/142135017-bc3cba9f-7271-4c31-8134-e793b4a24567.mov

### CODE REVIEW

File: components/automate-ui/src/app/pages/+compliance/+node-credentials/node-credential-details/node-credential-details.component.html [hunk 1]
```diff
@@ -30,7 +30,7 @@
       <section class="page-body" *ngIf="tabValue === 'details'">
         <chef-loading-spinner *ngIf="nodeDetailsLoading" size="50"></chef-loading-spinner>
         <ng-container *ngIf="!nodeDetailsLoading" >
-        
+
```
Comment (shaik80) [inline components/automate-ui/src/app/pages/+compliance/+node-credentials/node-credential-details/node-credential-details.component.html:33]: why we need this new line..?

File: components/automate-ui/src/app/pages/+compliance/+node-credentials/node-credential-details/node-credential-details.component.ts [hunk 2]
```diff
@@ -170,12 +170,31 @@ export class NodeCredentialDetailsScreenComponent implements OnInit, OnDestroy {
...
+        }
+}
 
   ngOnDestroy() {
     this.isDestroyed.next(true);
     this.isDestroyed.complete();
   }
+
+  checkError(field: string , type: any) {
+   return  ((type.get(field).hasError('required') || type.get(field).hasError('pattern')) && type.get(field).dirty) && type.get(field).value !== '';
```
Comment (shaik80) [inline components/automate-ui/src/app/pages/+compliance/+node-credentials/node-credential-details/node-credential-details.component.ts:198]: indentation fix, add one more space before return statement

**Review by shaik80 — COMMENTED** [review]
please fix the Pipeline issue

File: components/automate-ui/src/app/pages/+compliance/+node-credentials/node-credential-details/node-credential-details.component.html (hunk 1 above)
Comment (shivangi-singh1) [reply components/automate-ui/src/app/pages/+compliance/+node-credentials/node-credential-details/node-credential-details.component.html]: done

File: components/automate-ui/src/app/pages/+compliance/+node-credentials/node-credential-details/node-credential-details.component.ts (hunk 2 above)
Comment (shivangi-singh1) [reply components/automate-ui/src/app/pages/+compliance/+node-credentials/node-credential-details/node-credential-details.component.ts]: done

File: components/automate-ui/src/app/pages/+compliance/+node-credentials/node-credential-details/node-credential-details.component.ts [hunk 3]
```diff
@@ -170,12 +170,32 @@ export class NodeCredentialDetailsScreenComponent implements OnInit, OnDestroy {
...
+        data.type === 'ssh' ? this.sshForms
+        : data.type === 'winrm' ? this.winrmForms
+        : ''
+        ))) {
+          this.store.dispatch(new UpdateNodeCredential(this.nodeCredential));
+          if (data.type !== 'sudo') {
+            this.resetForm.controls[data.type]['controls']['username'].setValue('');
+          }
+          this.resetForm.controls[data.type]['controls']['password'].setValue('');
+        }
```
Comment (vivekshankar1) [inline components/automate-ui/src/app/pages/+compliance/+node-credentials/node-credential-details/node-credential-details.component.ts:189]:
```suggestion
    if (!this.checkError(
          'password',
          data.type === 'ssh' ? this.sshForms
          : data.type === 'winrm' ? this.winrmForms
          : this.sudoForms
        ) || (!this.checkError(
          'username',
          data.type === 'ssh' ? this.sshForms
          : data.type === 'winrm' ? this.winrmForms
          : ''
    ))) {
      this.store.dispatch(new UpdateNodeCredential(this.nodeCredential));
      if (data.type !== 'sudo') {
        this.resetForm.controls[data.type]['controls']['username'].setValue('');
      }
      this.resetForm.controls[data.type]['controls']['password'].setValue('');
    }
```

**Review by vivekshankar1 — APPROVED** [review]

**Review by shaik80 — APPROVED** [review]

**Review by vivek-yadav — APPROVED** [review]

**Pull request merged**